app/models/spree/active_sale_event.rb

```diff
@@ -56,6 +57,10 @@ def update_permalink
       self.permalink = self.name.parameterize if self.permalink.blank?
     end
 
+    def update_line_items
+      ::ActiveSale::LineItemUpdater.call(product_ids: self.product_ids) if self.discount_changed?
```

Comment:
@sokmesakhiev it would be more efficient if we do async with background job to update line item and send the active_sale_event id instead of product ids to avoid redis memory blown up.

```
CartLineItemUpdaterJob.perform_later(self.id)
```
line_item_updater is a good name thus it may be chosen in the future by spree or another gem so I think it would be great to specify a bit more context-oriented with cart thus I would choose cart_line_item_updater.